02-peer-protocol.md

```diff
@@ -559,6 +560,8 @@ A sending node:
         (witness program versions 1 through 16)
 
 A receiving node:
+  - if it receives multiple `shutdown` messages:
+    - SHOULD ignore all except the first one.
```

Comment:
I agree, I hesitated to suggest removing it as well.
That would be totally fine by me (and less bloat in the spec is always a win).

Reply:
> Mh! I don't like the idea to ignore messages here too because I'm not sure if we can check if a message arrives first or second right?

What do you mean? Of course you know what message arrived first as a receiver...? And since senders must not do it anyway, this can only happen for a buggy sender so you don't have to care much about the outcome, it's very likely that this will end up in a force-close anyway.

> we miss the possibility to catch this bug during the developing cycle without panic node in production

I think you're exaggerating the importance of this issue :slightly_smiling_face:

Reply:
>I think you're exaggerating the importance of this issue

Yeah, I know, but I'm just trying to find a reason to answer myself if all agree that this is a force close why not put it in the spec? I'm not trying to push in the spec repo, but I would like to know why. Just because I don't like to leave myself with pending questions.

Reply:
We don't all agree it must be a force-close, there are multiple behaviors in that case that are ok.

Eclair just ignores the other `shutdown` and will try doing a mutual close based on the script received in the first `shutdown`, which may end up in a force-close if the buggy sender changed the script (but may work otherwise).

So I don't want the spec to mandate that I should force-close here, but depending on the codebase, it could make sense for others to always force-close, so I don't want to mandate anything either to them. We must give some leeway to implementations on things that don't matter much such as this to avoid over-constraining them and giving them enough room to use different architectures.

Reply:
This make sense, thanks @t-bast